src/Take.Blip.Builder/Utils/SalesForce/SalesForceClient.cs

```diff
@@ -0,0 +1,156 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Take.Blip.Builder.Hosting;
+using Take.Blip.Builder.Models;
+using Take.Blip.Builder.Utils.SalesForce.Models;
+
+namespace Take.Blip.Builder.Utils.SalesForce
+{
+    public class SalesForceClient : ICrmClient
+    {
+        private readonly ILogger _logger;
+        private const string LEAD_ATTRIBUTE_KEY = "attributes";
+        private readonly IConfiguration _configuration;
+
+        public SalesForceClient(ILogger logger, IConfiguration configuration)
+        {
+            _logger = logger;
+            _configuration = configuration;
+        }
+
+        public async Task<AuthorizationResponse> GetAuthorizationAsync(CrmConfig crmConfig, string ownerId, CancellationToken cancellationToken)
+        {
+            var salesForceConfig = crmConfig.SalesForceConfig;
+            var body = new Dictionary<string, string>()
+            {
+                {SalesForceConstants.GRANT_TYPE_KEY, SalesForceConstants.GRANT_TYPE },
+                {SalesForceConstants.CLIENT_ID, salesForceConfig.ClientId },
+                {SalesForceConstants.CLIENT_SECRET, salesForceConfig.ClientSecret},
+                {SalesForceConstants.REFRESH_TOKEN, salesForceConfig.RefreshToken }
+            };
+
+            var responseToken = string.Empty;
```

Comment:
Since this will not be used outside the `try` you can declare it when calling `ReadAsStringAsync`.